instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/MetricAttribute.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+/**
+ * A class representing a metric attribute. It is responsible for extracting the attribute value (to
+ * be reported as a Measurement attribute), and for holding the corresponding attribute name to be
+ * used. Objects of this class are immutable.
+ */
+public class MetricAttribute {
+  private final String name;
+  private final MetricAttributeExtractor extractor;
+
+  public MetricAttribute(String name, MetricAttributeExtractor extractor) {
+    this.name = name;
+    this.extractor = extractor;
+  }
+
+  public String getAttributeName() {
+    return name;
+  }
+
+  String acquireAttributeValue(MBeanServer server, ObjectName objectName) {
+    return extractor.extractValue(server, objectName);
+  }
+
+  public static MetricAttributeExtractor fromConstant(String constantValue) {
+    return (a, b) -> {
+      return constantValue;
+    };
+  }
+
+  public static MetricAttributeExtractor fromObjectNameParameter(String parameterKey) {
+    if (parameterKey.isEmpty()) {
+      throw new IllegalArgumentException("Empty parameter name");
+    }
+    return (dummy, objectName) -> {
+      return objectName.getKeyProperty(parameterKey);
+    };
+  }
+
+  public static MetricAttributeExtractor fromBeanAttribute(String attributeName) {
+    return BeanAttributeExtractor.fromName(attributeName);
+  }
```

Comment:
we typically put these kinds of static creation methods inside the interface itself, `MetricAttributeExtractor`

Reply:
I'll move them